### DISCUSSION TITLE
Help with amico.Evaluation

### DISCUSSION THREAD

**Discussion by mmsignes** [daducci/AMICO#168] [Q&A] [answered]
Hello, 

I am trying to use the function amico.Evaluation in the following way: ae = amico.Evaluation("Studyname", subj), where subj is a variable with the name of the subject folder, so the script could run for all the subjects in "Studyname" folder. However, the function is not working. I tried to use it in the simplest way proposed in the wiki: ae = amico.Evaluation("Studyname", "Subject01"), where the arguments are the names of the different folders (study and first subject), but is not working either. Am I using the function in a wrong way? I would appreciate any help. Thanks!

Mar.

**Comment by daducci** [comment]
Hi,
Could you you post the output of the terminal as well as a screenshot of your folder atructure? Otherwise we cannot understand what is going on.

**Reply by mmsignes** [reply]
Hi,
My folder structure is the following:
…/Project/
	Noddi_script.py
	Study/
		S01/
		S02/
		…
The output of the terminal is:

<img width="536" alt="Screenshot 2023-09-12 at 10 33 54" src="https://github.com/daducci/AMICO/assets/144680488/84a96437-782f-4c25-8b01-2ae22aa9aeab">

I think this happens because it cannot find the subject's folder (as the same script it's running if I launch it inside of subject's folder).

Thank you.

**Reply by nightwnvol** [reply]
Hi @mmsignes,
this error occurs when you provide an incorrect file name or path to the `fsl2scheme` function. Please note that the function requires the paths to your `bval` and `bvec` files relative to your `Noddi_script.py` file.

e.g.
```python
fsl2scheme('Study/S01/eddy_dwi.bval', 'Study/S01/eddy_dwi.bvec')
```

**Reply by mmsignes** [reply]
Thank you for the reply. Then, what's **amico.Evaluation** for? I thought once you gave to this function the arguments of "study" and "subject", you would only have to indicate the name of bvals and bvecs files (for fsl2scheme function) but that the path was not necessary anymore.

Mar.

**Reply by nightwnvol** [reply, marked as answer]
You're right; after setting the `study_path` and `subject` arguments in your `Evaluation` object, you only have to specify the input file names for the methods of the `Evaluation` class, e.g. the `load_data` method.

```python
ae = amico.Evaluation("Study", "S01")
ae.load_data("S01_dwi.nii.gz", "eddy_dwi.scheme", "S01_mask.nii.gz")
```

`fsl2scheme` is not a method of the `Evaluation` class, so you have to specify the entire path of your `bval` and `bvec` files. This is because in a study, if the data were acquired with the same acquisition parameters, the subjects will share the same `bval` and `bvec` files. Therefore, one could generate a single scheme file and use it for all the subjects in the study.

I agree with you that this can be a bit confusing. Maybe we can consider a better implementation in future releases.

**Reply by mmsignes** [reply]
Thank you! Now is clear.

Mar.
